service/src/java/org/apache/hive/service/cli/CLIService.java

```diff
@@ -133,6 +134,15 @@ private void applyAuthorizationConfigPolicy(HiveConf newHiveConf) throws HiveExc
     ss.setIsHiveServerQuery(true);
     SessionState.start(ss);
     ss.applyAuthorizationPolicy();
+
+    // Add shutdown hook to close the session.
+    ShutdownHookManager.addShutdownHook(() -> {
```

Comment:
does it make sense to introduce something like:
```
ss.closeOnExit()
```
which will be reusable on the SessionState object level